test/unit/decorators/belongsTo.test.js

```diff
@@ -1,184 +1,158 @@
+    })
+
+    it('should throw error if related mapper is undefined', function () {
+      assert.throws(() => {
+        belongsTo(undefined, { localField: 'bar', foreignKey: 'fooId' })(mapper)
+      }, Error, '[new Relation:related] expected: Mapper or string, found: undefined\nhttp://www.js-data.io/v3.0/docs/errors#400')
+    })
+
+    it('should not throw error if related mapper is a string and "getRelation" option is a function', function () {
+      assert.doesNotThrow(() => {
+        belongsTo('another', { localField: 'bar', foreignKey: 'fooId', getRelation: () => anotherMapper })(mapper)
+      })
     })
-    const foo = store.add('foo', { id: 1 })
-    const bar = store.add('bar', { id: 1, foo_id: 1 })
-    assert.strictEqual(bar._foo, foo)
   })
 
-  it('should add property accessors to prototype of target and allow relation re-assignment using customizations', function () {
-    const store = new JSData.DataStore()
-    store.defineMapper('foo', {
-      relations: {
-        hasMany: {
-          bar: {
-            localField: 'bars',
-            foreignKey: 'fooId'
-          }
+  describe('when 2 way relation is defined (belongsTo + hasMany)', function () {
+    let foos, bars
+
+    beforeEach(function () {
+      store = createStore()
+      store.defineMapper('foo', {
+        relations: {
+          hasMany: createRelation('bar', { localField: 'bars', foreignKey: 'fooId' })
         }
-      }
-    })
-    store.defineMapper('bar', {
-      relations: {
-        belongsTo: {
-          foo: {
-            localField: '_foo',
-            foreignKey: 'fooId'
-          }
+      })
+      store.defineMapper('bar', {
+        relations: {
+          belongsTo: createRelation('foo', { localField: 'foo', foreignKey: 'fooId' })
         }
-      }
-    })
-    const foo = store.add('foo', { id: 1 })
-    const foo2 = store.add('foo', { id: 2 })
-    const bar = store.add('bar', { id: 1, fooId: 1 })
-    const bar2 = store.add('bar', { id: 2, fooId: 1 })
-    const bar3 = store.add('bar', { id: 3 })
-    assert.strictEqual(bar._foo, foo)
-    assert.strictEqual(bar2._foo, foo)
-    assert(!bar3._foo)
-    bar._foo = foo2
-    bar3._foo = foo
-    assert.strictEqual(bar._foo, foo2)
-    assert.strictEqual(bar2._foo, foo)
-    assert.strictEqual(bar3._foo, foo)
+      })
+      foos = store.add('foo', [{ id: 1 }, { id: 2 }])
+      bars = store.add('bar', [{ id: 1, fooId: 1 }, { id: 2, fooId: 1 }])
+    })
+
+    it('should add property accessors to prototype of target' , function () {
+      const Foo = store.getMapper('foo').recordClass
+      const Bar = store.getMapper('bar').recordClass
+
+      assert.isTrue(Foo.prototype.hasOwnProperty('bars'))
+      assert.isTrue(Bar.prototype.hasOwnProperty('foo'))
+    })
+
+    it('should automatically map relations when record is added to store', function () {
+      assert.deepEqual(foos[0].bars, bars)
+      assert.isTrue(bars.every(bar => bar.foo === foos[0]))
```

Comment:
Does Babel transpile uses of `Array.prototype.every` into ES5-compatible code? If not then we can't use it.